webextension/background.js

```diff
@@ -43,7 +44,41 @@ const themeManager = {
   }
 };
 
+const tabPageCounter = {
+  init() {
+    browser.tabs.onCreated.addListener(this.initTabCounter);
+    browser.tabs.onRemoved.addListener(this.sendTabCountAndDelete);
+    browser.webRequest.onCompleted.addListener(this.incrementTabCount, {urls: ["<all_urls>"], types: ["main_frame"]});
+  },
+
+  initTabCounter(tab) {
+    tabPageRequestCounter[tab.id] = {
+      "cookieStoreId": tab.cookieStoreId,
+      "pageRequests": 0
+    };
+  },
+
+  sendTabCountAndDelete(tab) {
+    browser.runtime.sendMessage({
+      method: "sendTelemetryPayload",
+      event: "page-requests-completed-per-tab",
+      userContextId: tabPageRequestCounter[tab].cookieStoreId,
+      pageRequestCount: tabPageRequestCounter[tab].pageRequests
+    });
+    delete tabPageRequestCounter[tab.id];
+  },
+
+  incrementTabCount(details) {
+    browser.tabs.get(details.tabId).then(tab => {
```

Comment:
Are we (singleArg) => {} or singleArg => {}. One of us will have to change. Seems we are inconsistent anyway :(